### PR TITLE
ci: Increase the dynamic port range to the maximum on native Windows

### LINKED ISSUE #18548: Tests fail intermittently on Windows (appveyor) "OSError: [WinError 10048] Only one usage of each socket address (protocol/network address/port) is normally permitted"
Example:

* https://ci.appveyor.com/project/DrahtBot/bitcoin/builds/31988561#L4009

### PR DESCRIPTION
**Pull request by hebasto**
Fixes #18548.

The solution suggested in https://github.com/bitcoin/bitcoin/issues/18548#issuecomment-923944390.

Also more functional tests added to the native Windows task.

### CODE REVIEW

**Review by vincenzopalazzo — APPROVED** [review]
ack https://github.com/bitcoin/bitcoin/pull/23089/commits/097ac74fd22f706fb7d8d652dadea844dc8198b9

**Comment by MarcoFalke** [comment]
concept ACK 097ac74fd22f706fb7d8d652dadea844dc8198b9

File: .cirrus.yml [hunk 1]
```diff
@@ -147,6 +147,10 @@ task:
     - python test\util\test_runner.py
     - python test\util\rpcauth-test.py
   functional_tests_script:
+    # Increase the dynamic port range to the maximum allowed value to mitigate "OSError: [WinError 10048] Only one usage of each socket address (protocol/network address/port) is normally permitted".
+    # See: https://docs.microsoft.com/en-us/biztalk/technical-guides/settings-that-can-be-modified-to-improve-network-performance
+    - netsh int ipv4 set dynamicport tcp start=1025 num=64511
+    - netsh int ipv6 set dynamicport tcp start=1025 num=64511
```
Comment (MarcoFalke) [inline .cirrus.yml:153]: does this mean they might collide with the p2p and rpc ports from `[TEST_RUNNER_PORT_MIN, TEST_RUNNER_PORT_MIN + 2 * range]` (`[14000, 24000]`)?

**Comment by MarcoFalke** [comment]
Maybe this allows to revert fab98992043f47fa7240d7c1217920d0c4f783a2 some day?

**Review by darosior — COMMENTED** [review]
Concept ACK. Most of the latest PRs are currently failing CI because of this.

File: .cirrus.yml (hunk 1 above)
Comment (hebasto) [reply .cirrus.yml:153]: Yes, if an ephemeral port by accident belongs to that range a collision is possible. But this range limits only ports of P2P and RPC connections. Our test framework uses ports out of this range in the `feature_bind_extra.py` and `feature_proxy.py`. The former is skipped on Windows, but the latter still could be a source of port collisions.

**Pull request merged**